engines/bastion_katello/app/assets/javascripts/bastion_katello/products/bulk/products-bulk-http-proxy-modal.controller.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @ngdoc object
+ * @name  Bastion.products.controller:ProductsBulkHttpProxyModalController
+ *
+ * @requires $scope
+ * @requires $state
+ * @requires $sce
+ * @requires translate
+ * @requires ProductBulkAction
+ * @requires CurrentOrganization
+ * @requires Notification
+ * @requires $uibModalInstance
+ * @requires bulkParams
+ * @requires globalContentProxy
+ *
+ * @description
+ *   A controller for providing bulk action functionality to the products page.
+ */
+angular.module('Bastion.products').controller('ProductsBulkHttpProxyModalController',
+    ['$scope', '$state', '$sce', 'translate', 'HttpProxy', 'ProductBulkAction', 'CurrentOrganization', 'Notification', '$uibModalInstance', 'bulkParams', 'globalContentProxy',
+        function ($scope, $state, $sce, translate, HttpProxy, ProductBulkAction, CurrentOrganization, Notification, $uibModalInstance, bulkParams, globalContentProxy) {
+            var success, error, globalName;
+
+            if (globalContentProxy) {
+                globalName = translate("Global Default") + " (" + globalContentProxy + ")"
+            } else {
+                globalName = translate("Global Default (None)")
+            }
+
+            $scope.proxy_options = {http_proxy_policy: 'global_default_http_proxy',
+                                    proxy_id: undefined};
+            $scope.policies = [ {name: globalName, label: 'global_default_http_proxy'},
+                                {name: translate("No HTTP Proxy"), label: 'none'},
+                                {name: translate("Use specific HTTP Proxy"), label: 'use_selected_http_proxy'}]
+
+            $scope.proxies = [];
```

Comment:
you'll probably be doing something similar to this, we may want to make it sharable, similar to this: https://github.com/Katello/katello/blob/master/engines/bastion_katello/app/assets/javascripts/bastion_katello/common/module-stream-actions.service.js